### PR TITLE
feat: index referral events

### LINKED ISSUE #28: Create referral Indexer and querys
Brief https://app.skiff.com/file/6801dd4a-525e-496b-bcfa-4463d0b12b59

### PR DESCRIPTION
**Pull request by irisdv**
This PR closes #28 

It indexes `on_claim` and `on_commission` events from the referral contract. 

To test it on the devnet you can use script `deploy_devnet` here : https://github.com/starknet-id/referral_contract/pull/2 (Explanation to run the script is located in the README file in this PR)

### CODE REVIEW

**Review by fricoben — APPROVED** [review]
LGTM waiting for thomas cause he is the chad

File: indexer/listener.py [hunk 1]
```diff
@@ -107,10 +107,17 @@ def add_filter(contract, event):
...
+            "on_claim",
+            "on_commission",
+        ]:
+            add_filter(self.conf.referral_contract, starknet_id_event)
+
         return IndexerConfiguration(
             filter=filter,
             starting_cursor=starknet_cursor(self.conf.starting_block),
-            finality=DataFinality.DATA_STATUS_PENDING,
+            finality=DataFinality.DATA_STATUS_ACCEPTED if self.conf.is_devnet is True else DataFinality.DATA_STATUS_PENDING,
```
Comment (Th0rgal) [inline indexer/listener.py:120]: From now on you can keep using ACCEPTED only, PENDING was removed in 0.12.0 (which should be on mainnet soon).

Comment (Th0rgal) [inline indexer/listener.py:120]: From now on you can keep using ACCEPTED only, PENDING was removed in 0.12.0 (which should be on mainnet soon).

**Review by Th0rgal — APPROVED** [review]
lgtm

**Pull request merged**